### DISCUSSION TITLE
How can i do? (press enter focus next input widget)

### DISCUSSION THREAD

**Discussion by burhansvural** [Textualize/textual#3943] [General]
I have 8 input widgets on the screen. I want to focus the next input widget when I press the enter key instead of the tab key. How can I do this?

**Comment by davep** [comment]
You could focus the next widget when `Input.Submitted` is posted, as one approach:

```python
from textual import on
from textual.app import App, ComposeResult
from textual.widgets import Input

class FocusNextOnSubmutApp(App[None]):

    def compose(self) -> ComposeResult:
        for n in range(10):
            yield Input(placeholder=f"This is input {n}")

    @on(Input.Submitted)
    def go_next(self) -> None:
        self.screen.focus_next()

if __name__ == "__main__":
    FocusNextOnSubmutApp().run()
```

**Reply by burhansvural** [reply]
Thanks dave. it's ok.

**Reply by burhansvural** [reply]
Dave look at this code.... I don't want focused selection list, i want only focused next input widget press enter key 

from textual import on
from textual.app import App, ComposeResult
from textual.widgets import Input, SelectionList

class FocusNextOnSubmutApp(App[None]):

    CSS_PATH = "selection_list.tcss"
    
    def compose(self) -> ComposeResult:
        for n in range(10):
            yield Input(placeholder=f"This is input {n}")
            if n==5:
                yield SelectionList[int](  
                    ("Falken's Maze", 0, True),
                    ("Black Jack", 1),
                    ("Gin Rummy", 2),
                    ("Hearts", 3),
                    ("Bridge", 4),
                    ("Checkers", 5),
                    ("Chess", 6, True),
                    ("Poker", 7),
                    ("Fighter Combat", 8, True),
        )

    @on(Input.Submitted)
    def go_next(self) -> None:
        self.screen.focus_next()
        
    def on_mount(self) -> None:
        self.query_one(SelectionList).border_title = "Shall we play some games?"

if __name__ == "__main__":
    FocusNextOnSubmutApp().run()

**Reply by davep** [reply]
[`focus_next` will take a selecator](https://textual.textualize.io/api/screen/#textual.screen.Screen.focus_next), so you could tell it exactly what it should look for next to focus. For example:

```python
from textual import on
from textual.app import App, ComposeResult
from textual.widgets import Input, Checkbox

class FocusNextOnSubmutApp(App[None]):

    def compose(self) -> ComposeResult:
        for n in range(10):
            if n == 5:
                yield Checkbox("We'll skip this!")
            else:
                yield Input(placeholder=f"This is input {n}")

    @on(Input.Submitted)
    def go_next(self) -> None:
        self.screen.focus_next(Input)

if __name__ == "__main__":
    FocusNextOnSubmutApp().run()
```

**Reply by burhansvural** [reply]
Perfect. Thank you very much Dave. I loved Textual.

**Reply by davep** [reply]
Glad to hear you're enjoying working with it. 🙂

**Reply by burhansvural** [reply]
In the 90s, I was using the clipper programming language when writing programs on the DOS operating system. Now, with textual, it is very nice and at the same time it has much more advanced capabilities. I really enjoy writing programs for the DOS operating system. Thank you very much. 🙂

**Reply by davep** [reply]
Funnily enough for most of the 90s I built systems with CA-Clipper too; used to hang out in `comp.lang.clipper` a lot, used to maintain the VFAQ for it and also took over the FAQ later on. I was also involved in [Harbour](https://harbour.github.io/) in the early days.

**Reply by burhansvural** [reply]
dave help me pls.

# I have a this product list
    
    lstProducts=[
        ["apple",0.50],
        ["banana",0.75],
        ["potato chips",0.45],
        ["soda",1.25],
        ["milk",4.50]        
    ]
    
    # product , quantity, price, total amount
    # product , quantity, price, total amount
    # product , quantity, price, total amount
    #                          +______________________
    #                          total amount to be paid
                              
    # This example will simulate shopping and paying at a supermarket. 
    # Can you explain how I can do this simply with an example?

**Reply by burhansvural** [reply]
It's..... Perfect.... Nice to meet you and see some of the old software developers.

**Reply by burhansvural** [reply]
It's been exactly 8 years since I stopped dealing with the software business. I'm back now and the things I've seen are great and I've been trying to prepare myself to work in the software industry again for about 20 days.

**Reply by burhansvural** [reply]
I'm getting very excited right now when I see Textual's capabilities.

**Reply by TomJGooding** [reply]
@burhansvural I would recommend working through the [Textual tutorial](https://textual.textualize.io/tutorial/) if you haven't done so already. The tutorial provides a great introduction to important concepts such as custom widgets. reactive attributes and messages that it looks like you'll need for your application.

Textual's documentation really is excellent, so after finishing the tutorial I would encourage also checking out the [Guide](https://textual.textualize.io/guide/) for an in-depth reference.

Of course if there's something you need help with or just unsure about, as you've found the Textual community is happy to help!

**Comment by burhansvural** [comment]
I am working now. Thank you very much, Tom.

TomJGooding ***@***.***>, 2 Oca 2024 Sal, 20:25 tarihinde
şunu yazdı:
